hack/install_kustomize.sh

```diff
@@ -25,11 +25,54 @@ unset CDPATH
 
 where=$PWD
 
+function getTagName () {
```

Comment:
In trying to figure out how this works (I'm not great at bash!) I discovered it seems to be taken from [StackOverflow](https://stackoverflow.com/questions/4023830/how-to-compare-two-strings-in-dot-separated-version-format-in-bash). Since that uses a CC BY-SA 4.0 license, I don't think we can accept it--Kustomize uses Apache 2.0.

That said, we are dealing with a well-known set of tags here, and both they and our input are guaranteed (by the regex at L73) to have three digits, which makes the task much simpler than the general case. In fact, I'd suggest going with the most naive solution of literally checking the list of exceptions, e.g.:

```bash
function getTagName () {
    local requested=$1
    # The complete list of tags published before we started prefixing
    local unprefixedTags=(v1.0.0 v1.0.1 v1.0.10 v1.0.11 v1.0.2 v1.0.3 v1.0.4 v1.0.5 v1.0.6 v1.0.7 v1.0.8 v1.0.9 v2.0.0 v2.0.1 v2.0.2 v2.0.3 v2.1.0 v3.0.0 v3.0.1 v3.0.2 v3.0.3 v3.1.0 v3.2.0 v3.3.0 v3.3.1)

    for tag in ${unprefixedTags[@]}; do
        if [[ $tag = "v$requested" ]]; then
            echo "v$requested"
            return 0
        fi
    done

    echo "kustomize/v$requested"
    return 0
}
```